### ISSUE TITLE
inconsistent clearing of per-link domain list

### ISSUE THREAD

**Issue by rvka** [systemd/systemd#24706]
### systemd version the issue has been seen with

systemd-250.8-1.fc36.x86_64

### Used distribution

Fedora 36

### Linux kernel version used

5.19.8-200.fc36.x86_64

### CPU architectures issue was seen on

x86_64

### Component

systemd-resolved

### Expected behaviour you didn't see

When a new domain list is specified for an interface, it should replace the old one

### Unexpected behaviour you saw

When a new domain list is specified for an interface, only one of the old domains is removed from the list

### Steps to reproduce the problem

```
$ resolvectl domain tun0
Link 38 (tun0):
$ sudo resolvectl domain tun0 foo.com bar.com baz.com
$ resolvectl domain tun0
Link 38 (tun0): bar.com baz.com foo.com
$ sudo resolvectl domain tun0 x.com y.com
### expect x.com y.com
$ resolvectl domain tun0
Link 38 (tun0): bar.com baz.com x.com y.com
$ sudo resolvectl domain tun0 ''
### expect empty list
$ resolvectl domain tun0
Link 38 (tun0): baz.com x.com y.com
### verify that the problem is in systemd-resolved, not in resolvectl
$ sudo gdbus call  --system --dest org.freedesktop.resolve1 --object-path /org/freedesktop/resolve1/link/_338 --method org.freedesktop.resolve1.Link.SetDomains '[]'
()
### expect empty list
$ resolvectl domain tun0
Link 38 (tun0): x.com y.com
```

**Comment by yuwata** [comment]
IIRC, the issue was recently fixed. Is it possible to reproduce the issue with v251 or the current git HEAD?

**Comment by yuwata** [comment]
Yeah, already fixed by 1196b6a2e6. Closing.
